portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,15 +35,62 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private List<String> names;
+
+  private List<String> messages;
+
   @Override
   public void init(){
     names = new ArrayList<>();
     names.add("Obi");
+
+    messages = new ArrayList<>();
+    messages.add("Hello! How are you");
+    //messages.add("You got this!");
+    //messages.add("Nice to meet you.");
   }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String name = names.get(0);
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello " + name + ". Welcome to JavaServlets");
+
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String whatSaid = (String) entity.getProperty("comment");
+      String mood = (String) entity.getProperty("mood");
+      long timestamp = (Long) entity.getProperty("timestamp");
+      
+      Comment comment = new Comment(id, whatSaid, mood, timestamp);
+      //messages.add(comment);
```

Comment:
I htink you need to uncomment this if you want it to work.  otherwise you create comment & then never use it.

Reply:
also if messages lives longer than on request I think you'll keep adding the same comments over and over.